fleet/comms.go

```diff
@@ -423,10 +451,34 @@ func (svc fleetCommsService) handleCapabilities(thingID string, channelID string
 	agent.AgentMetadata["backends"] = capabilities.Backends
 	agent.AgentMetadata["orb_agent"] = capabilities.OrbAgent
 	agent.AgentTags = capabilities.AgentTags
-	err := svc.agentRepo.UpdateDataByIDWithChannel(context.Background(), agent)
+
+	err := svc.checkVersion(orb.GetMinVersion(), capabilities.OrbAgent.Version, &agent)
+	if err != nil {
+		return err
+	}
+
+	err = svc.agentRepo.UpdateDataByIDWithChannel(context.Background(), agent)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (svc fleetCommsService) checkVersion(minVersion string, agentVersion string, agent *Agent) error {
+	mVersion, err := version.NewVersion(minVersion)
 	if err != nil {
 		return err
 	}
+	aVersion, err := version.NewVersion(agentVersion)
+	if err != nil {
+		return err
+	}
+
+	if aVersion.LessThan(mVersion) {
+		// TODO dispatch a Agent Stop RPC to the agent to inform that it's not possible to communicate with the fleet
+		svc.NotifyAgentStop(agent.MFChannelID)
```

Comment:
we can send a Reason of "The orb-agent version is too old to connect to the control plane. Minimum required version: {}" with mVersion